pkg/client/cache/delta_fifo.go

```diff
@@ -452,6 +456,27 @@ func (f *DeltaFIFO) Replace(list []interface{}, resourceVersion string) error {
 	return nil
 }
 
+// Resync will send a sync event for each item
+func (f *DeltaFIFO) Resync() error {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
+	for _, k := range f.knownObjects.ListKeys() {
```

Comment:
Actually, I was completely wrong here.
I think it's enough to iterate over knownObjects.

We can ignore f.items, because those things are already queue for processing, so adding an "artificial Sync" event for them doesn't change anything, because those would be processed anyway.